feat(frontend/backend): routes detail page ss-268

timosh-yura opened the pull request:
Added:

- routes reducer and getRoute action
- routes page at "/app/routes/{:id}"
- routes get api to whiteRoutes

Changed backend:

- routes description row increased varchar length to 2k
 
<img width="1484" height="805" alt="image" src="https://github.com/user-attachments/assets/ca551b6d-f6f9-4e34-8ac1-a7e95b0d9431" />

h4vefundieyoung commented on apps/frontend/src/pages/route/libs/components/images-section/styles.module.css:
```diff
@@ -0,0 +1,43 @@
+	gap: 16px;
+}
+
+.container img {
```
can't be used according to the quality criteria

h4vefundieyoung commented on apps/frontend/src/pages/route/libs/components/images-section/styles.module.css:
```diff
@@ -0,0 +1,43 @@
+}
+
+.primary-image {
+	height: 416px;
```
are you sure that we should use that fixed height?

liza-veis replied:
i also think that here we can use max height instead with height: 100%

h4vefundieyoung commented on apps/frontend/src/pages/route/libs/components/images-section/styles.module.css:
```diff
@@ -0,0 +1,43 @@
+}
+
+.secondary-image {
+	height: 200px;
```
same

timosh-yura replied:
Last time Lisa covered this topic, it was suggested that fixed height may be ok for the image height

timosh-yura replied:
I`m not sure how to style it otherwise

liza-veis commented on apps/frontend/src/pages/route/libs/components/images-section/styles.module.css:
```diff
@@ -0,0 +1,44 @@
+.container {
```
Let's use a grid instead. In short, we need 1 column which takes two rows and 2 columns where 1 image takes one row and the other one another row. For mobile screens, let's just keep them one row and add overflow auto, to make it scrollable to the right, here is an example
https://codepen.io/Tkrish/pen/xxZpwMK

timosh-yura replied:
Please describe more detailed

liza-veis replied:
I mean, currently we have 2 rows with images. On smaller screens, it won't fit well, so we need to somehow adjust it. One way is to make all of them 1 row (the same size) and make them horizontally scrollable. At least in my imagination, it seems like it will look good here, but if you have other suggestions for mobile view, you can implement them. Also, let's make a common container for the page on bigger screens smaller, to maintain the same ratio between the large image and the smaller ones

timosh-yura commented:
Added:

- whiteRoutes path behaves like a regex

Chagned

- description not on the same column as a primary image, is this ok?

-  max-width for big screens

<img width="1728" height="846" alt="image" src="https://github.com/user-attachments/assets/4725ca20-2e7f-47d8-9f73-53a939769609" />

- Added horizontal scroll for mobile screens

<img width="778" height="859" alt="image" src="https://github.com/user-attachments/assets/f315c95b-08cf-4503-ae5b-2522eab1af34" />

kalinkaaaa14 commented on apps/frontend/src/pages/route/route-details.tsx:
```diff
@@ -0,0 +1,60 @@
+	return <Loader />;
+};
+
+export { RouteDetails as Route };
```
```suggestion
export { RouteDetails };
```

kalinkaaaa14 commented on packages/shared/src/modules/routes/libs/types/routes-response-dto.type.ts:
```diff
@@ -1,4 +1,13 @@
-type RoutesResponseDto = {
+type RouteGetAllItemResponseDto = {
+	id: number;
+	name: string;
+	pois: {
+		id: number;
+		visitOrder: number;
+	}[];
+}[];
+
+type RouteGetByIdResponseDto = {
```
these types should be placed in different files, and do not forget about filenames

kalinkaaaa14 commented on package.json:
```diff
@@ -69,5 +69,8 @@
 		"commit-msg": "npm run git:hooks:commit"
+	},
+	"dependencies": {
+		"path-to-regexp": "8.2.0"
```
Seems this lib is used only on backend, so install it not in the root package please, but on the backend

timosh-yura commented on apps/frontend/src/modules/routes/routes-api.ts:
```diff
@@ -0,0 +1,37 @@
+		id: number,
+	): Promise<APIResponse<RouteGetByIdResponseDto>> {
+		const response = await this.load<APIResponse<RouteGetByIdResponseDto>>(
+			this.getFullEndpoint(`${RoutesApiPath.ROOT}${id.toString()}`, {}),
```
indeed
  26:49  error  Invalid type "number" of template literal expression  @typescript-eslint/restrict-template-expressions

timosh-yura commented:
Updated, "routes" store would be in the same module or I should also rename every filename in "routes" dir to singular?

https://github.com/user-attachments/assets/65804f2e-4f54-423e-957a-9cc5fc6f6edb

liza-veis commented on apps/frontend/src/pages/route-description/libs/components/images-section/images-section.tsx:
```diff
@@ -0,0 +1,34 @@
+const ImagesSection = (): React.JSX.Element => {
+	return (
+		<section className={styles["container"]}>
+			<div className={styles["primary-wrapper"]}>
+				<img
+					alt="Main point of interest of the route"
+					className={styles["primary-image"]}
+					src={image1}
+				/>
+			</div>
+
+			<div className={styles["secondary-images-wrapper"]}>
+				<img
+					alt="Secondary point of interest of the route"
+					className={styles["secondary-image"]}
+					src={image2}
+				/>
+				<img
+					alt="Secondary point of interest of the route"
+					className={styles["secondary-image"]}
+					src={image3}
+				/>
+			</div>
```
We don't need all the wrappers around images; this can be done easily with a grid. I tried to play a bit with it and was able to achieve this result. Pay attention to how responsiveness works here, and obviously, we need to adjust styles according to conventions. This was just a draft example

https://codepen.io/liza-veis/pen/bNVYEJz

Also let's rename this component to ImageGallery

liza-veis commented on apps/frontend/src/pages/route-description/route-details.tsx:
```diff
@@ -0,0 +1,60 @@
+		);
+	}
+
+	return <Loader />;
```
let's return the component the last one, and loader if status is pending

liza-veis commented on apps/frontend/src/pages/route-description/styles.module.css:
```diff
@@ -0,0 +1,34 @@
+.container {
```
Let's leave a to-do comment here that these styles should be moved to the common layout

liza-veis commented on apps/frontend/src/pages/route-description/route-details.tsx:
```diff
@@ -0,0 +1,60 @@
+		const { description, name } = data as RouteGetByIdResponseDto;
+
+		return (
+			<main>
+				<Header
+					actions={[{ label: "Sign in", to: AppRoute.SIGN_IN }]}
+					user={authenticatedUser}
+				/>
+				<div className={styles["container"]}>
+					<label className={styles["label"]}>{name}</label>
+					<ImagesSection />
+					<p className={styles["description"]}>{description}</p>
+				</div>
+			</main>
```
Let's get rid of this unnecessary wrapper. We will move this to the common layout later. Also header should be h1, not label. And FYI, each page should have an H1 title, even if it's hidden, it should be present in the layout
```suggestion
               <>
				<Header
					actions={[{ label: "Sign in", to: AppRoute.SIGN_IN }]}
					user={authenticatedUser}
				/>
			<main className={styles["container"]}>
					<h1 className={styles["title"]}>{name}</label>
					<ImagesSection />
					<p className={styles["description"]}>{description}</p>
			</main>
			</>
```

liza-veis commented on packages/shared/src/modules/routes/libs/types/routes-response-get-all-dto.ts:
fileName should end with .type.ts

liza-veis reviewed (changes requested):
Overall looks good, here are a few comments 🐱

kalinkaaaa14 commented on apps/backend/src/db/migrations/20250810152620_alter_routes_description_length.ts:
```diff
@@ -0,0 +1,26 @@
+const DEFAULT_DESCRIPTION_LENGTH = 255;
+
+async function down(knex: Knex): Promise<void> {
+	await knex("routes")
```
run these calls in a transaction please

timosh-yura commented:
what`s the convo on filenames?
not sure what`s wrong with "routes-response-get-all-dto.type.ts", changed to "routes-response-get-all-item-dto.type.ts"

timosh-yura commented:
Do I need to change something for the merge?

h4vefundieyoung commented on apps/backend/src/modules/routes/routes.controller.ts:
```diff
@@ -7,14 +7,15 @@ import {
 import { type Logger } from "~/libs/modules/logger/logger.js";
 
-import { RouteApiPath } from "./libs/enums/enums.js";
+import { RoutesApiPath } from "./libs/enums/enums.js";
```
I'm not really sure, these naming stuff getting confusing, but I believe we should name enums in singular form. So it should be RouteApiPath

h4vefundieyoung commented on apps/backend/src/modules/routes/routes.controller.ts:
```diff
@@ -428,7 +429,7 @@ class RoutesController extends BaseController {
 	public async findById(
 		options: APIHandlerOptions<{ params: { id: string } }>,
-	): Promise<APIHandlerResponse<RoutesResponseDto>> {
+	): Promise<APIHandlerResponse<RouteGetByIdResponseDto>> {
```
I'm not really sure, the naming stuff is getting confusing, but it's routes everywhere, shouldn't we follow this pattern?

timosh-yura replied:
I originally picked up naming pattern from points-of-interest (plural)
should i rename every file to singular?

kalinkaaaa14 replied:
@timosh-yura no, fix only places that are highlighted in this pr, we will fix the naming through the project in separate ticket

h4vefundieyoung commented on packages/shared/src/modules/routes/libs/types/routes-response-get-all-item-dto.type.ts:
@kalinkaaaa14 so we're going to use route and not routes as convention, right?

kalinkaaaa14 commented on packages/shared/src/modules/routes/libs/types/route-update-request.dto.ts:
route-patch-request-dto.type.ts

kalinkaaaa14 commented on packages/shared/src/modules/routes/libs/types/routes-get-id-response-dto.type.ts:
route-get-by-id-response-dto.type.ts

kalinkaaaa14 commented on packages/shared/src/modules/routes/libs/types/routes-request-dto.type.ts:
route-create-request-dto.type.ts

timosh-yura commented:
*fixed
